multiapps-controller-persistence/src/main/java/org/cloudfoundry/multiapps/controller/persistence/services/ProcessLoggerProvider.java

```diff
@@ -1,50 +1,78 @@
+import org.cloudfoundry.multiapps.controller.persistence.model.ImmutableOperationLogEntry;
+import org.cloudfoundry.multiapps.controller.persistence.model.OperationLogEntry;
+import org.cloudfoundry.multiapps.controller.persistence.query.providers.SqlOperationLogQueryProvider;
 import org.flowable.engine.delegate.DelegateExecution;
 
 @Named("processLoggerProvider")
 public class ProcessLoggerProvider {
 
     static final String LOG_LAYOUT = "#2.0#%d{yyyy MM dd HH:mm:ss.SSS}#%d{XXX}#%p#%c#%n%X{MsgCode}#%X{CSNComponent}#%X{DCComponent}##%X{DSRCorrelationId}#%X{Application}#%C#%X{User}#%X{Session}#%X{Transaction}#%X{DSRRootContextId}#%X{DSRTransaction}#%X{DSRConnection}#%X{DSRCounter}#%t##%X{ResourceBundle}#%n%m#%n%n";
-
+    private static final LevelMatchFilter DEBUG_FILTER = LevelMatchFilter.newBuilder()
+                                                                         .setLevel(Level.DEBUG)
+                                                                         .setOnMatch(Filter.Result.ACCEPT)
+                                                                         .setOnMismatch(Filter.Result.ACCEPT)
+                                                                         .build();
     private static final String PARENT_LOGGER = "com.sap.cloud.lm.sl.xs2";
     private static final String DEFAULT_LOG_NAME = "OPERATION";
-    private static final String DEFAULT_LOG_DIR = "logs";
     private static final String LOG_FILE_EXTENSION = ".log";
+    private final DataSource dataSource;
 
     private final Map<String, ProcessLogger> loggersCache = new ConcurrentHashMap<>();
+    private final LoggerContext loggerContext = (LoggerContext) LogManager.getContext(false);
+
+    @Inject
+    public ProcessLoggerProvider(DataSource dataSource) {
+        this.dataSource = dataSource;
+    }
+
+    public ProcessLoggerProvider() {
+        this.dataSource = null;
+    }
 
     public ProcessLogger getLogger(DelegateExecution execution) {
         return getLogger(execution, DEFAULT_LOG_NAME);
     }
 
-    public ProcessLogger getLogger(DelegateExecution context, String logName) {
-        return getLogger(context, logName, loggerContext -> PatternLayout.newBuilder()
-                                                                         .withPattern(LOG_LAYOUT)
-                                                                         .withConfiguration(loggerContext.getConfiguration())
-                                                                         .build());
+    public ProcessLogger getLogger(DelegateExecution execution, String logName) {
+        return getLogger(execution, logName, loggerContextDel -> PatternLayout.newBuilder()
+                                                                              .withPattern(LOG_LAYOUT)
+                                                                              .withConfiguration(loggerContextDel.getConfiguration())
```

Comment:
do you need the lambda?

Reply:
the same pattern is used in the xsa that's why i kept it here